pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerManager.java

```diff
@@ -106,14 +134,21 @@ public CompletableFuture<Versioned<LedgerMetadata>> createLedgerMetadata(long le
             return FutureUtil.failedFuture(new BKException.BKMetadataSerializationException(ioe));
         }
 
-        CompletableFuture<Versioned<LedgerMetadata>> future = store.put(getLedgerPath(ledgerId), data, Optional.of(-1L))
-                .thenApply(stat -> new Versioned(metadata, new LongVersion(stat.getVersion())));
-        future.exceptionally(ex -> {
-            log.error("Failed to create ledger {}: {}", ledgerId, ex.getMessage());
-            return null;
-        });
+        CompletableFuture<Versioned<LedgerMetadata>> promise = new CompletableFuture<>();
```

Comment:
Just a small concern:
I prefer the original chain-style `CompletableFuture`. it has two benefits than current:
1. If we use a chain-style future, we don't need to create a separate `CompletableFuture`.
2. In the current approach, it has risks cause the future can not complete yet. for example, throw an exception before we complete the `promise`. (But this method looks fine.

Please let me know if you have any good suggestions. thanks a lot!

Reply:
yes, agree.  Why create a new future here?